new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -436,104 +743,222 @@ public static boolean createSecret(String secretName,
+    String name = secret.getMetadata().getName();
+
+    KubernetesApiResponse<V1Secret> response = secretClient.delete(namespace, name);
+
+    if (!response.isSuccess()) {
+      throw new ApiException("Failed to delete secret '" + name + "' from namespace: "
+          + namespace + " with HTTP status code: " + response.getHttpStatusCode());
+    }
 
-    if (status.getCode() == 200 || status.getCode() == 202) {
-      // status code 200 = OK, 202 = Accepted
-      return true;
+    if (response.getObject() != null) {
+      logger.info(
+          "Received after-deletion status of the requested object, will be deleting "
+              + "secret in background!");
     }
 
-    return false;
+    return true;
   }
 
   // --------------------------- pv/pvc ---------------------------
 
   /**
-   * Delete the Kubernetes Persistent Volume
    *
-   * @param pvName the name of the Persistent Volume
-   * @return true on success, false otherwise
-   * @throws ApiException - if Kubernetes client API call fails
+   * @param persistentVolume - V1PersistentVolume object containing Kubernetes persistent volume
+   *     configuration data
+   * @return true if successful
+   * @throws ApiException - missing required configuration data, if Kubernetes request fails or
+   *     unsuccessful
    */
-  public static boolean deletePv(String pvName) throws ApiException {
-    V1DeleteOptions deleteOptions = new V1DeleteOptions();
+  public static boolean createPv(V1PersistentVolume persistentVolume) throws ApiException {
+    if (persistentVolume == null) {
+      throw new ApiException(
+          "Parameter 'persistentVolume' is null when calling createPv()");
+    }
 
-    V1Status status = coreV1Api.deletePersistentVolume(
-        pvName, // persistent volume (PV) name
-        pretty, // pretty print output
+    V1PersistentVolume pv = coreV1Api.createPersistentVolume(
+        persistentVolume, // persistent volume configuration data
+        PRETTY, // pretty print output
         null, // indicates that modifications should not be persisted
-        0, // duration in seconds before the object should be deleted
-        false, // Should the dependent objects be orphaned
-        "Foreground", // Whether and how garbage collection will be performed
-        deleteOptions
+        null // fieldManager is a name associated with the actor
     );
 
-    if (status.getCode() == 200 || status.getCode() == 202) {
-      // status code 200 = OK, 202 = Accepted
-      return true;
+    return true;
+  }
+
+  /**
+   * @param persistentVolumeClaim - V1PersistentVolumeClaim object containing Kubernetes
+   *     persistent volume claim configuration data
+   * @return true if successful
+   * @throws ApiException - missing required configuration data, if Kubernetes request fails or
+   *     unsuccessful
+   */
+  public static boolean createPvc(V1PersistentVolumeClaim persistentVolumeClaim) throws ApiException {
+    if (persistentVolumeClaim == null) {
```

Comment:
Is the test expected to create the PVC Object ? or do we need another POJO object to hold the values of PVC details ?

Reply:
Yes, the test is expected to create the PVC object.

Reply:
Currently, my understanding is that the test would create the Kubernetes PVC model object using a fluent style builder since it already exists as part of the api.